src/mpid/ch4/netmod/ofi/ofi_types.h

```diff
@@ -397,6 +350,23 @@ typedef struct {
     char kvsname[MPIDI_KVSAPPSTRLEN];
     char pname[MPI_MAX_PROCESSOR_NAME];
     int port_name_tag_mask[MPIR_MAX_CONTEXT_MASK];
+
+    /* Capability settings */
+#ifdef MPIDI_OFI_ENABLE_RUNTIME_CHECKS
```

Comment:
In our discussion here, we came up with roughly something like this:

#if defined(PROVIDER_1) || (PROVIDER_2)
#define MPIDI_OFI_ENABLE_TAGGED 1
#else
#define MPIDI_OFI_ENABLE_TAGGED (capabilities[0].tagged)
#endif

init_function()
{
    fi_check_capabilities(FI_NULL, &capabilities);
    /* maybe do some pruning here to remove providers that do not support our
     * mandatory set of capabilities required by mpich */
    assert(MPIDI_OFI_ENABLE_TAGGED == capabilities[0].tagged);
}